### ISSUE TITLE
🐛 Bug Report: awsalb auth provider does not work correctly for ALBs provisioned in AWS GovCloud regions

### ISSUE THREAD

**Issue by rosshillery** [backstage/backstage#24020]
### 📜 Description

Using the awsalb auth provider does not work correctly for resources provisioned in AWS GovCloud regions ("us-gov-west-1", "us-gov-east-1")

Specifically, it fails to pull verified public key for a JWT that was generated.

Within the `https://github.com/backstage/backstage/blob/master/plugins/auth-backend-module-aws-alb-provider/src/helpers.ts`

on line [83](https://github.com/backstage/backstage/blob/master/plugins/auth-backend-module-aws-alb-provider/src/helpers.ts#L83) the URL for public key retrieval is forced to the commercial endpoints

### 👍 Expected behavior

Based on the input region, auth provider should update the URL that is used to obtain the public key to be correct for AWS GovCloud regions.

AWS provides updated URL locations for load balancers that are provisioned in GovCloud regions as listed in these [help docs](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/listener-authenticate-users.html)

The AWS GovCloud endpoints should be...

```
https://s3-us-gov-west-1.amazonaws.com/aws-elb-public-keys-prod-us-gov-west-1/key-id
https://s3-us-gov-east-1.amazonaws.com/aws-elb-public-keys-prod-us-gov-east-1/key-id
```

Instead of the hard-coded 

`https://public-keys.auth.elb.region.amazonaws.com/key-id`

### 👎 Actual Behavior with Screenshots

When inputting an ALB that is located in either the us-gov-west-1 or us-gov-east-1 regions, on authentication, JWT processing returns a FetchError ENOTFOUND due to an attempt to lookup a non-existent URL.

Example `app-config.yaml`

```
auth:
  providers: 
    awsalb:
      issuer: "https://login.microsoftonline.us/<tenant_id>/v2.0"
      region: "us-gov-west-1"
```

```
{"level":"error","message":"Request failed with status 500 Exception occurred during JWT processing: FetchError: request to https://public-keys.auth.elb.us-gov-west-1.amazonaws.com/<REDACTED> failed, reason: getaddrinfo ENOTFOUND  │
│ public-keys.auth.elb.us-gov-west-1.amazonaws.com","service":"backstage","stack":"Error: Exception occurred during JWT processing: FetchError: request to https://public-keys.auth.elb.us-gov-west-1.amazonaws.com/<REDEACTED> failed, r │
│ eason: getaddrinfo ENOTFOUND public-keys.auth.elb.us-gov-west-1.amazonaws.com\n    at Object.authenticate (/app/node_modules/@backstage/plugin-auth-backend-module-aws-alb-provider/dist/index.cjs.js:152:13)\n    at process.processTicksAndRejections (node:in │
│ ternal/process/task_queues:95:5)\n    at async Object.refresh (/app/node_modules/@backstage/plugin-auth-node/dist/index.cjs.js:945:40)","type":"errorHandler"}

```

### 👟 Reproduction steps

In a fresh backstage installation execute the following steps....

Follow [this tutorial](https://github.com/backstage/backstage/blob/master/contrib/docs/tutorials/aws-alb-aad-oidc-auth.md) for additional reference

1. After updating backstage backend and frontend to utilize awsalb auth provider
2. Insert provider that is located in either the "us-gov-west-1" or "us-gov-east-1" region
3. Attempt to login
4. You will receive JWT ENOTFOUND error

### 📃 Provide the context for the Bug.

This bug prevents our organization from utilizing the awsalb auth provider for our backstage instance that is running behind an ALB provisioned in the AWS GovCloud region

### 🖥️ Your Environment

```
OS:   Linux 5.10.102.1-microsoft-standard-WSL2 - linux/x64
node: v20.11.1
yarn: 1.22.22
cli:  0.25.2 (installed)
backstage:  1.23.0

Dependencies:
  @backstage/app-defaults                                          1.5.3
  @backstage/backend-app-api                                       0.5.14, 0.6.2, 0.6.3-next.0
  @backstage/backend-common                                        0.20.2, 0.21.6, 0.21.7-next.0
  @backstage/backend-dev-utils                                     0.1.4
  @backstage/backend-openapi-utils                                 0.1.9
  @backstage/backend-plugin-api                                    0.6.16, 0.6.17-next.0
  @backstage/backend-tasks                                         0.5.21, 0.5.22-next.0
  @backstage/backend-test-utils                                    0.3.7-next.0
  @backstage/catalog-client                                        1.6.3
  @backstage/catalog-model                                         1.4.5
  @backstage/cli-common                                            0.1.13
  @backstage/cli-node                                              0.2.4
  @backstage/cli                                                   0.25.2, 0.26.3-next.0
  @backstage/config-loader                                         1.7.0, 1.8.0-next.0
  @backstage/config                                                1.2.0
  @backstage/core-app-api                                          1.12.3
  @backstage/core-compat-api                                       0.2.3
  @backstage/core-components                                       0.13.10, 0.14.3
  @backstage/core-plugin-api                                       1.9.1
  @backstage/e2e-test-utils                                        0.1.1
  @backstage/errors                                                1.2.4
  @backstage/eslint-plugin                                         0.1.6
  @backstage/frontend-plugin-api                                   0.6.3
  @backstage/integration-aws-node                                  0.1.12
  @backstage/integration-react                                     1.1.25
  @backstage/integration                                           1.10.0-next.0, 1.9.1
  @backstage/plugin-api-docs                                       0.11.3
  @backstage/plugin-app-backend                                    0.3.64
  @backstage/plugin-app-node                                       0.1.16
  @backstage/plugin-auth-backend-module-atlassian-provider         0.1.8, 0.1.9-next.0
  @backstage/plugin-auth-backend-module-aws-alb-provider           0.1.8
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.11, 0.2.12-next.0
  @backstage/plugin-auth-backend-module-github-provider            0.1.13, 0.1.14-next.0
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.13, 0.1.14-next.0
  @backstage/plugin-auth-backend-module-google-provider            0.1.13, 0.1.14-next.0
  @backstage/plugin-auth-backend-module-microsoft-provider         0.1.11, 0.1.12-next.0
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.13, 0.1.14-next.0
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider      0.1.10-next.0, 0.1.9
  @backstage/plugin-auth-backend-module-oidc-provider              0.1.7, 0.1.8-next.0
  @backstage/plugin-auth-backend-module-okta-provider              0.0.10-next.0, 0.0.6, 0.0.9
  @backstage/plugin-auth-backend                                   0.21.3, 0.22.3, 0.22.4-next.0
  @backstage/plugin-auth-node                                      0.4.11, 0.4.12-next.0
  @backstage/plugin-auth-react                                     0.0.3
  @backstage/plugin-catalog-backend-module-gitlab                  0.3.14
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.14
  @backstage/plugin-catalog-backend                                1.21.0
  @backstage/plugin-catalog-common                                 1.0.22
  @backstage/plugin-catalog-graph                                  0.4.3
  @backstage/plugin-catalog-import                                 0.10.9
  @backstage/plugin-catalog-node                                   1.11.0, 1.11.1-next.0
  @backstage/plugin-catalog-react                                  1.11.2
  @backstage/plugin-catalog                                        1.18.2
  @backstage/plugin-events-node                                    0.3.2
  @backstage/plugin-github-actions                                 0.6.14
  @backstage/plugin-home-react                                     0.1.11
  @backstage/plugin-home                                           0.6.2
  @backstage/plugin-kubernetes-backend                             0.15.3
  @backstage/plugin-kubernetes-common                              0.7.5
  @backstage/plugin-kubernetes-node                                0.1.10
  @backstage/plugin-kubernetes-react                               0.3.3
  @backstage/plugin-kubernetes                                     0.11.8
  @backstage/plugin-org                                            0.6.23
  @backstage/plugin-permission-common                              0.7.13
  @backstage/plugin-permission-node                                0.7.27, 0.7.28-next.0
  @backstage/plugin-permission-react                               0.4.21
  @backstage/plugin-proxy-backend                                  0.4.14
  @backstage/plugin-scaffolder-backend-module-azure                0.1.8
  @backstage/plugin-scaffolder-backend-module-bitbucket-cloud      0.1.6
  @backstage/plugin-scaffolder-backend-module-bitbucket-server     0.1.6
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.2.6
  @backstage/plugin-scaffolder-backend-module-gerrit               0.1.8
  @backstage/plugin-scaffolder-backend-module-gitea                0.1.6
  @backstage/plugin-scaffolder-backend-module-github               0.2.6
  @backstage/plugin-scaffolder-backend-module-gitlab               0.3.2
  @backstage/plugin-scaffolder-backend                             1.22.3
  @backstage/plugin-scaffolder-common                              1.5.1
  @backstage/plugin-scaffolder-node                                0.4.2
  @backstage/plugin-scaffolder-react                               1.8.3
  @backstage/plugin-scaffolder                                     1.19.2
  @backstage/plugin-search-backend-module-catalog                  0.1.21
  @backstage/plugin-search-backend-module-pg                       0.5.25
  @backstage/plugin-search-backend-module-techdocs                 0.1.21
  @backstage/plugin-search-backend-node                            1.2.20
  @backstage/plugin-search-backend                                 1.5.6
  @backstage/plugin-search-common                                  1.2.11
  @backstage/plugin-search-react                                   1.7.9
  @backstage/plugin-search                                         1.4.9
  @backstage/plugin-tech-radar                                     0.6.13
  @backstage/plugin-techdocs-backend                               1.10.3
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.8
  @backstage/plugin-techdocs-node                                  1.12.2
  @backstage/plugin-techdocs-react                                 1.2.2
  @backstage/plugin-techdocs                                       1.10.3
  @backstage/plugin-user-settings                                  0.8.4
  @backstage/release-manifests                                     0.0.11
  @backstage/test-utils                                            1.5.3
  @backstage/theme                                                 0.5.2
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.7
  ```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by cftad** [comment]
Hi @rosshillery,

Thanks for the detail in the report and testing the fix - #24053 has now been merged and will be part of the `1.26.0` release 🎉

**Comment by freben** [comment]
Closing as fixed! 🥳

**Comment by rosshillery** [comment]
> Hi @rosshillery,
> 
> Thanks for the detail in the report and testing the fix - #24053 has now been merged and will be part of the `1.26.0` release 🎉

Thanks so much for working the fix! Glad to get this in, we are working on upgrading to 1.26!
